hoot-core/src/main/cpp/hoot/core/io/OsmChangeWriter.cpp

```diff
@@ -40,56 +44,50 @@ void OsmChangeWriter::write(QIODevice &d, ChangeSetProviderPtr cs)
   writer.writeAttribute("version", "0.6");
   writer.writeAttribute("generator", "hootenanny");
 
-  Change::ChangeType last = Change::Unknown;
-
+  //Make the changeset have only one set of entries per change type.  Also for readability,
+  //sort the set of changes within each change type by element type.
+  QMap<Change::ChangeType, QMap<ElementType::Type, QVector<ConstElementPtr> > > elementsByChangesetTypeAndElementType;
```

Comment:
If that is the case then this is the right fix. We can circle back and deal with memory bound issues later as needed.